lte/gateway/c/core/oai/tasks/s1ap/s1ap_state.cpp

```diff
@@ -177,6 +177,7 @@ void remove_ues_without_imsi_from_ue_id_coll() {
 
     // for each ue comp_s1ap_id in eNB->ue_id_coll, check if it has an S1ap
     // ue_context, if not delete it
+    mme_ue_id_no_imsi_list.clear();
```

Comment:
mme_ue_id_no_imsi_list is already empty (line no. 159). Is it necessary to call clear()?

Reply:
Yes it's necessary because the vector, mme_ue_id_no_imsi_list holds the UEs for which IMSI is not allocated. This list is constructed for every eNB. 
Here we are traversing for all eNBs in loop, before it proceeds for next eNB, the list must cleared.